src/domain/data/index.ts

```diff
@@ -0,0 +1,44 @@
+export class DataProvider {
+  private readonly encoder = new TextEncoder();
+  constructor(
+    private readonly textLen: number,
+    private readonly patternLen: number
+  ) {}
```

Comment:
Maybe we should move some initialization code to the body of the constructor

Reply:
Could you clarify please?

Reply:
The `private readonly` keywords already incorporate the passed arguments as the class instance's private members.

Reply:
I had the "= new TextEncoder();" on my mind. Up to you if you'd like to move it or have it as it is.

Reply:
Oh I understand now. The GitHub code snippet shows the lines below. I get your point, but this is a specific case - a `readonly` member. 

> Readonly properties must be initialized at their declaration or in the constructor.

Source: [https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier](https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier)